src/DualQuaternion.jl

```diff
@@ -6,29 +6,29 @@ struct DualQuaternion{T<:Real} <: Number
   norm::Bool
 end
 
-DualQuaternion(q0::Quaternion, qe::Quaternion, n::Bool = false) =
+DualQuaternion(q0::Quaternion, qe::Quaternion, n::Bool = q0.norm) =
```

Comment:
This does not seem to be consistent with how `norm` is defined for `DualQuaternion`. `norm` returns a `Dual` number, and the `DualQuaternion` is normalized when the real part of the norm is 1 and the infinitesimal part is 0.

Reply:
I think we only want to set `norm=true` when its value can be determined from either the types being provided or from the value of other `norm` fields, which I don't think we can do here.

Reply:
I have zero knowledge of dual quaternions, hence my mistake. It appears this has veered wildly beyond the scope of both this PR and my understanding and I'll revert my change.

Off-topic discussion: wouldn't your statement imply that `n::Bool = q0.norm & iszero(qe)` would be the proper initialization? More importantly, doesn't any requirement on the infinitesimal part imply that `normalize(::DualQuaternion)` and friends are incorrect in setting `norm=true` without examining the infinitesimal? According to [https://en.wikipedia.org/wiki/Dual_quaternion#Norm](url), however, it appears to be slightly less restrictive than what you have claimed. It says that a dual quaternion `A+eB` is unit when `(AA*,AB*+BA*)=(1,0)`.

Reply:
TBH, I don't have much experience with dual quaternions either, so I can't really answer the questions without digging into the code and the theory a bit more. My point was just that the change would create inconsistencies in the code base.